dexmaker/src/test/java/com/google/dexmaker/stock/ProxyBuilderTest.java

```diff
@@ -879,4 +881,65 @@ public void setFakeResult(Object result) {
             fakeResult = result;
         }
     }
+
+    public static class TestOrderingClass {
+        public int returnsInt() {
+            return 0;
+        }
+
+        public String returnsString() {
+            return "string";
+        }
+
+        public boolean returnsBoolean() {
+            return false;
+        }
+
+        public double returnsDouble() {
+            return 1.0;
+        }
+
+        public Object returnsObject() {
+            return new Object();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testMethodsGeneratedInDeterministicOrder() throws Exception {
+        Field mapField = ProxyBuilder.class
+                .getDeclaredField("generatedProxyClasses");
+        mapField.setAccessible(true);
+
+        // Grab the static methods array from the original class.
+        Class<?> clazz1 = proxyFor(TestOrderingClass.class).buildProxyClass();
```

Comment:
No, ProxyBuilder.buildProxyClass() returns the proxy class. ProxyBuilder.build() returns an instance of the class.